src/ui/svgcomponent.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 The boardgame.io Authors
+ *
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Models } from './svg';
+
+class SvgComponent extends React.Component {
+  static propTypes = {
+    component: PropTypes.element.isRequired,
+    _center: PropTypes.object,
+    style: PropTypes.any,
+    onClick: PropTypes.func,
+  };
+
+  static defaultProps = {
+    style: { fill: '#ff0000' },
+  };
+
+  render() {
+    let tx = 0,
+      ty = 0;
+    if (this.props._center !== undefined) {
+      tx = this.props._center.x;
+      ty = this.props._center.y;
+    }
+
+    return (
+      <g onClick={this.props.onClick} transform={`translate(${tx}, ${ty})`}>
+        {this.props.component}
```

Comment:
i think here you want to actually render the passed component? you could set it to a variable above the return statement ( and just make sure its uppercase like a react component should be) like `const SvgComponent = this.props.component` then on this line render it `<SvgComponent />`

Reply:
Shouldn't it render the component even as the code stands now?

Reply:
The way it stands the component is just being referenced here, an array of components can be rendered like that ie: `{this.props.children}` but I’m pretty sure that a single component needs to be rendered like `<Component />` while currently it’s just a reference `Component`, I only mentioned this because I thought it wasn’t working yet? If it is working currently then then I guess i am probably Mistaken.